src/SFML/Audio/SoundSource.cpp

```diff
@@ -206,4 +223,45 @@ SoundSource::Status SoundSource::getStatus() const
     return Stopped;
 }
 
+void SoundSource::setEffect(const SoundEffect* effect)
```

Comment:
It's still not clear to me why effect can't be a const ref here, same as `sf::SoundBuffer` is passed by const ref to `sf::Sound::setBuffer`. To remove effect, we can add `removeEffect` function, which will be an explicit way of saying "I don't want this effect anymore".

Reply:
This has been addresed in [1b41eb0](https://github.com/SFML/SFML/pull/1708/commits/1b41eb0821fb56436836929b7e25fd36781b63a6) 🙂